common/models/user.js

```diff
@@ -27,6 +28,36 @@ function getAboutProfile({
   };
 }
 
+function getStatsProfile({
+  username,
+  challengeMap = {}
+}) {
+  return {
+    username,
+    stats: challengeMapToStats(challengeMap)
+  };
+}
+
+// @return array of
+// { date: $commit_date,
+//   url: $repo_url, (optional)
+//   hash: $commit_hash
+// }
+function challengeMapToStats(obj) {
+  var commit = [];
+  Object.keys(obj).map((k) => {
+    if (obj[k].hasOwnProperty('completedDate') &&
+        obj[k].hasOwnProperty('files')) {
+      commit.push({
+        date: obj[k].completedDate,
+        hash: crypto.createHash('sha1')
```

Comment:
Though it may be working when changing to another shasum algorithm, but strictly speaking, it is wrong when sending to `git-stats --record`. Because right now SHA-1 is still the shasum string generator using by `git`.